### PR TITLE
add feign client

### PR DESCRIPTION
**Pull request by florinv23**

### CODE REVIEW

File: src/main/resources/application-docker-local.properties [hunk 1]
```diff
@@ -21,15 +21,13 @@ spring.datasource.password = root
...
-cart.service.url = http://localhost:8083
-product.service.url = http://localhost:8081
 user.service.url = http://localhost:8080
 
 # Email configurations
 spring.mail.host = smtp.hostinger.com
 spring.mail.port = 587
 spring.mail.username = noreply@springwebstore.com
-spring.mail.password = ${HOSTINGER_EMAIL_PASS}
+spring.mail.password = Brut6vdWNgK-?nq
```
Comment (GeorgeT88) [inline src/main/resources/application-docker-local.properties:30]: would be nice not to have the actual password here 😸 
maybe set it as a local env variable

File: src/main/resources/application-docker-local.properties [hunk 2]
```diff
@@ -66,6 +64,10 @@ cron.expression= 0 15 10 ? * *
 spring.main.banner-mode=off 
 spring.output.ansi.enabled=ALWAYS
 
-#disable Eureka Client
-eureka.client.enabled= true
+eureka.client.serviceUrl.defaultZone=http://localhost:8761/eureka
```
Comment (GeorgeT88) [inline src/main/resources/application-docker-local.properties:67]: some configuration comments would be muchly appreciated

File: src/main/java/com/boot/user/service/EmailService.java [hunk 3]
```diff
@@ -22,14 +22,15 @@
...
-@AllArgsConstructor
+@RequiredArgsConstructor
 public class EmailService {
 
     private static final String CONFIRMATION_EMAIL_TEMPLATE = "/templates/email-template.vm";
 
     private static final String RESET_PASSWORD_EMAIL_TEMPLATE = "/templates/reset-password-email-template.vm";
 
-    AppConfig appConfig;
+    @Value("${user.service.url}")
```
Comment (GeorgeT88) [inline src/main/java/com/boot/user/service/EmailService.java:32]: isn't there any possibility to use in this case feign option aswell?

**Review by GeorgeT88 — CHANGES_REQUESTED** [review]

**Comment by GeorgeT88** [comment]
also, there are 4 sonar code smells 😄 
and AppConfig.java should be ignored from sonar analyse

File: src/main/java/com/boot/user/service/EmailService.java (hunk 3 above)
Comment (florinv23) [reply src/main/java/com/boot/user/service/EmailService.java:32]: using feign for this is not possible. we could point it to the gateway url

**Review by GeorgeT88 — APPROVED** [review]

**Pull request merged**